src/Tests/Tests.fs

```diff
@@ -123,3 +123,33 @@ let ``Should generate valid Uatp`` () =
     card |> should startWith "1"
     card |> should haveLength 15
     card |> luhn |> should be LuhnCheck
+
+
+[<Theory>]
+[<InlineData(DinersClubInternationalLengthOptions.Fourteen, 14)>]
+[<InlineData(DinersClubInternationalLengthOptions.Fifteen, 15)>]
+[<InlineData(DinersClubInternationalLengthOptions.Sixteen, 16)>]
+[<InlineData(DinersClubInternationalLengthOptions.Seventeen, 17)>]
+[<InlineData(DinersClubInternationalLengthOptions.Eightteen, 18)>]
+[<InlineData(DinersClubInternationalLengthOptions.Nineteen, 19)>]
+let ``Should generate valid DinersClubInternational`` length expectedLength =
+    let card = Cardizer.NextDinersClubInternational length
+    card |> should startWith "36" 
+    card |> should haveLength expectedLength
+    card |> luhn |> should be LuhnCheck
+
+[<Fact>]
+let ``Should generate valid DinersClubUsAndCanada``  =
+    let card = Cardizer.NextDinersClubUsAndCanada()
+    card |> should startWith "54"
+    card |> should haveLength 16
+    card |> luhn |> should be LuhnCheck
+
+
+[<Fact>]
+let ``Should generate valid Diners``  =
+    let card = Cardizer.NextDinersClub()
+    let start = card.ToString().Substring(0, 2) |> int
+    let prefixInRange = start = 36 || start = 54 
+    prefixInRange |> should be True
```

Comment:
Hello,
Some points:
* card is already a `string` no need to do `.ToString()`. 
* you dont have to cast into an int if you compare to string

You could inspire yourself from the Verve card. In our case it would give:
```fsharp
    let card = Cardizer.NextDinersClub()
    let start = card.Substring(0, 2)
    let prefixIsValid = start = "36" || start = "54" 
    prefixIsValid |> should be True
    card |> luhn |> should be LuhnCheck
```